test/cli/unclosed-quote/test.out

```diff
@@ -0,0 +1,392 @@
+    end
+  end
+
+  def after<<todo method>>(&<blk>)
+    <emptyTree>
+  end
+end
+
+--------------------------------------------------------------------------
+
+# typed: false
+
+def before; end
+
+def main
+  %w(we are the people
+
+  'foo
+
+  %r(foo
+
+end
+
+def after
+end
+
+
+----- ./unclosed_quote_6.rb ---------------------------------------
+
+./unclosed_quote_6.rb:7: escape sequence meets end of file https://srb.help/2001
+     7 |    xyz = "#{}
+                  ^
+
+./unclosed_quote_6.rb:16: unexpected token "end of file" https://srb.help/2001
+    16 |
+    17 |
+
+./unclosed_quote_6.rb:7: Hint: this string literal might be missing a closing `"` at the end of the line https://srb.help/2003
+     7 |    xyz = "#{}
+                      ^
+
+./unclosed_quote_6.rb:9: Changing the type of a variable is not permitted in loops and blocks https://srb.help/7001
+     9 |      xyz = nil
+                    ^^^
+  Existing variable has type: `String`
+  Attempting to change type to: `NilClass`
+
+  Autocorrect: Done
+    ./unclosed_quote_6.rb:7: Replaced with `T.let("#{}, T.nilable(String))`
+     7 |    xyz = "#{}
+                  ^^^^
+Errors: 4
+class <emptyTree><<C <root>>> < (::<todo sym>)
+  def before<<todo method>>(&<blk>)
+    <emptyTree>
+  end
+
+  class <emptyTree>::<C A><<C <todo sym>>> < (::<todo sym>)
+    def foo<<todo method>>(&<blk>)
+      begin
+        xyz = ::<Magic>.<string-interpolate>(nil)
+        1.times() do ||
+          xyz = nil
+        end
+      end
+    end
+  end
+
+  def after<<todo method>>(&<blk>)
+    <emptyTree>
+  end
+end
+
+--------------------------------------------------------------------------
+
+# typed: true
+
+def before; end
+
+class A
+  def foo
+    xyz = T.let("#{}, T.nilable(String))
+    1.times do
+      xyz = nil
+    end
+  end
+end
+
+def after
+end
+
+
+----- ./unclosed_quote_7.rb ---------------------------------------
+
+./unclosed_quote_7.rb:9: escape sequence meets end of file https://srb.help/2001
+     9 |  x = "unclosed string
+              ^
+
+./unclosed_quote_7.rb:12: unexpected token "end of file" https://srb.help/2001
+    12 |def after; end
+    13 |
+
+./unclosed_quote_7.rb:6: Hint: this string literal might be missing a closing `"` at the end of the line https://srb.help/2003
+     6 |  x = "first line
+                         ^
+
+./unclosed_quote_7.rb:7: Hint: this string literal might be missing a closing `"` at the end of the line https://srb.help/2003
+     7 |  second line"
+                      ^
+
+./unclosed_quote_7.rb:9: Hint: this string literal might be missing a closing `"` at the end of the line https://srb.help/2003
+     9 |  x = "unclosed string
+                              ^
+Errors: 5
+class <emptyTree><<C <root>>> < (::<todo sym>)
+  def before<<todo method>>(&<blk>)
+    <emptyTree>
+  end
+
+  def foo<<todo method>>(&<blk>)
+    begin
+      x = "first line"
+      <self>.second(<self>.line(""))
+      x = "unclosed string"
```

Comment:
This is an interesting case, to show that in our single-line strings mode, something which previously would have parsed as

```
 x = "first line\n  second line"
 x = <emptyTree>::<C <ErrorNode>>
```

now parses as

```
 x = "first line"
 <self>.second(<self>.line(""))
 x = "unclosed string"
```

Arguably the new parse result is still better: the old parse result did not also include the `def after; end` method definition.

But depending on the contents of the first, multi-line string, it's almost certain that the continuation lines of the string literal parse as valid ruby code (like the `second(line(""))` parse result here). So adding this strategy is very likely to interfere with partial parse results in such files.